src/main/java/org/tndata/android/compass/activity/SnoozeActivity.java

```diff
@@ -50,32 +62,97 @@ protected void onResume(){
     }
 
     @Override
-    protected void onResumeFragments(){
-        super.onResumeFragments();
-        mTimePickerDialog.getDialog().setOnKeyListener(this);
-    }
-
-    @Override
     public void onBackPressed(){
         finish();
     }
 
     @Override
-    public void onTimeSet(RadialTimePickerDialog radialTimePickerDialog, int i, int i1){
-        finish();
+    public void onItemClick(AdapterView<?> parent, View view, int position, long id){
+        if (position == 0){
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTimeInMillis(calendar.getTimeInMillis()+3600*1000);
+            snooze(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH)+1,
+                    calendar.get(Calendar.DAY_OF_MONTH), calendar.get(Calendar.HOUR_OF_DAY),
+                    calendar.get(Calendar.MINUTE));
+        }
+        else if (position == 1){
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTimeInMillis(calendar.getTimeInMillis()+24*3600*1000);
+            snooze(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH)+1,
+                    calendar.get(Calendar.DAY_OF_MONTH), calendar.get(Calendar.HOUR_OF_DAY),
+                    calendar.get(Calendar.MINUTE));
+        }
+        else if (position == 2){
+            //Start the date picker
+            Calendar calendar = Calendar.getInstance();
+            CalendarDatePickerDialog datePickerDialog = CalendarDatePickerDialog.newInstance(this,
+                    calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH),
+                    calendar.get(Calendar.DAY_OF_MONTH));
+
+            datePickerDialog.show(getSupportFragmentManager(), "SnoozeDate");
```

Comment:
Since the snooze option is still in the API, do you want to replace the above with integer values (and maybe have a different `snooze` signature), or do you want to keep the date generation?